docs/bundler.md

```diff
@@ -0,0 +1,112 @@
+of the **Gemfile.lock** will be ignored. **Gemfile** is only necessary when parsing the **Gemfile.lock**
+to pre-fetch all dependencies specified in your project's **Gemfile.lock**. Please note, that we don't
+use any Bundler commands to pre-fetch dependencies. Instead, we use native mechanisms from the Ruby standard
+library to parse the **Gemfile.lock** and download the dependencies to the specified output directory.
+
+## Usage
+
+Run the following command in your terminal to pre-fetch your project's dependencies.
+The command will download all dependencies specified in the **Gemfile.lock** to the specified output directory.
+
+```bash
+cachi2 fetch-deps --source ./path-to-your-ruby-project --output ./cachi2-output bundler
+```
+
+### Configuration
+
+Bundler uses an unorthodox system when dealing with configuration options. The highest precedence
+is given to the config file, and then to the environment variables. See the official documentation:
+<https://bundler.io/v2.5/man/bundle-config.1.html#DESCRIPTION>>
+
+The order of precedence for Bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+Since the local configuration takes higher precedence than the environment variables
+(except `BUNDLE_APP_CONFIG`), we need to set a few Bundler configuration options to make the build
+work by copying the configuration file and setting the environment variables below. Then, we change
+the `BUNDLE_APP_CONFIG` environment variable to point to the copied/updated configuration file.
+
+This is an unfortunate limitation of Bundler, and we hope that Bundler will change this behavior in
+the future. It may be a problem if you have multiple packages in your repository with different
+settings. In that case, you may have to adjust the build phase accordingly.
+
+**BUNDLE_CACHE_PATH**: The directory that Bundler will place cached gems in when running
+`bundle package`, and that Bundler will look in when installing gems. Defaults to `vendor/cache`.
+
+**BUNDLE_DEPLOYMENT**: Disallow changes to the **Gemfile**. When the **Gemfile** is changed
+and the lockfile has not been updated, running Bundler commands will be blocked.
+
+**BUNDLE_NO_PRUNE**: Whether Bundler should leave outdated gems unpruned when caching.
+
+To create the configuration file, run the following command.
+You should see a log message that the file was created successfully.
+
+```bash
+cachi2 inject-files --for-output-dir /tmp/cachi2-output cachi2-output
+```
+
+Lastly, you need to set the `BUNDLE_APP_CONFIG` environment variable to point to the copied configuration file.
+
+```bash
+cachi2 generate-env --output ./cachi2.env --for-output-dir /tmp/cachi2-output ./cachi2-output
+```
+
+```bash
+# cat cachi2.env
+export BUNDLE_APP_CONFIG=/tmp/cachi2-output/bundler/config_override
+```
+
+### Hermetic build
+
```

Comment:
```
In this section you can find a sample Dockerfile to build a Ruby application.
```

Reply:
Why ? The section does not even have 30 lines.

Reply:
The draft version I reviewed did not have the introductory part in this section, this one has it, so not applicable any more.